File: Array/57.Insert_Interval/C++/57.Insert_Interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>> & intervals, vector<int> & newInterval) 
    {
        //=================================================//
        // Input type:                                     //
        //   - vector<vector<int>> ref. (intervals vector) //
        //   - vector<int> ref. (newInterval vector)       //
        // Return type:                                    //
        //   - vector<vector<int>> (result vector)         //
        //=================================================//

        /*Initialize*/
        ///// Length of intervals vector /////
        int len_intervals = intervals.size();

        ///// Record traversal pointer /////
        int record_trav_ptr = 0;

        ///// Result vector /////
        vector<vector<int>> res_vect;


        /*Greedy-based loop traversal with sorted, updated vector*/
        ///// Step 1: Sorted, Updated vector with new element /////
        intervals.push_back(newInterval); //Update, Record

        (len_intervals++); //Update, Accumulate

        sort(intervals.begin(), intervals.end(), [&](const vector<int> & interval_1, const vector<int> & interval_2)
        {
            return (interval_1[0] < interval_2[0]);

        }); //Sorted-operations

        ///// Step 2: Looped-traversal with sorted, updated vector /////
        while (record_trav_ptr < len_intervals) //Whole
        {
            int fst_val = (intervals[record_trav_ptr])[0]; int scnd_val = (intervals[record_trav_ptr])[1]; //Firts, Second value

            while (((record_trav_ptr + 1) < len_intervals) && ((intervals[(record_trav_ptr + 1)])[0] <= scnd_val)) //Part
            {
                ///// Check if the current indexed-value matched conditions or not /////
                if ((intervals[(record_trav_ptr + 1)])[1] <= scnd_val) { ; } 
                else { scnd_val = (intervals[(record_trav_ptr + 1)])[1]; } //Keep updating, overwriting

                (record_trav_ptr++); //Keep updating, traversing

            } //Part

            res_vect.push_back({fst_val, scnd_val}); //Keep updating, recording

            (record_trav_ptr++); //Keep updating, traversing

        } //Whole

        return res_vect;
    }
};

File: Array/57.Insert_Interval/Python/57.Insert_Interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of intervals array #####
        len_intervals = len(intervals)

        ##### Record traversal pointer #####
        record_trav_ptr = 0

        ##### Result array #####
        res_arry = []


        #######################################################
        #Greedy-based loop traversal with sorted/updated array
        ##### Step 1: Sorted/Updated array with new interval/element #####
        intervals.append(newInterval) #Update/Record

        intervals.sort() #Sorted operations

        len_intervals += 1 #Update/Accumulate

        ##### Step 2: Looped-traversal with sorted/updated array #####
        while (record_trav_ptr < len_intervals):
            fst_val, scnd_val = (intervals[record_trav_ptr])[0], (intervals[record_trav_ptr])[1] #Firts, Second value

            while (((record_trav_ptr + 1) < len_intervals) and ((intervals[(record_trav_ptr + 1)])[0] <= scnd_val)):

                ##### Check if the current indexed-value matched conditions or not #####
                if ((intervals[(record_trav_ptr + 1)])[1] <= scnd_val):
                    pass
                else:
                    scnd_val = (intervals[(record_trav_ptr + 1)])[1] #Keep updating/overwriting

                record_trav_ptr += 1 #Keep updating/traversing

            res_arry.append([fst_val, scnd_val]) #Keep updating/recording

            record_trav_ptr += 1 #Keep updating/traversing

        return res_arry
